compiler/src/middle_end/matchcomp.re

```diff
@@ -375,13 +375,7 @@ module MatchTreeCompiler = {
       );
     | Fail =>
       /* FIXME: We need a "throw error" node in ANF */
```

Comment:
Even though I made this trap instead of moving on silently, I left this note here because that's what needs to happen to throw `MatchFailure` instead of just `unreachable`.